src/app/fundraisings/[fundraisingId]/jars/JarsList/AddJarDialog/AddJarDialog.tsx

```diff
@@ -42,48 +60,70 @@ export const AddJarDialog = ({
 
   const handleSubmit = async (formData: FormData) => {
     const url = formData.get('url') as string;
-    const owner = formData.get('ownerName') as string;
-    const color = formData.get('jarColor') as string;
+    const owner = formData.get('owner-name') as string;
+    const color = formData.get('jar-color') as string;
+    const goal = formData.get('goal') as string;
+    const otherIncoms = formData.get('other-incoms') as string;
+    const isFinished = formData.get('is-finished') as string;
 
-    const existingJar = jars.find((jar) => {
+    const existingJar = jars?.find((jar) => {
       return jar.url === url || jar.ownerName === owner;
     });
-
-    if (existingJar) {
+    // temporary solution. add an event handler when editing
+    if (existingJar && !isEditMode) {
       setErrorText(`Така банка вже є у ${existingJar.ownerName}`);
       return;
     }
 
-    const existingColor = jars.some((jar) => {
+    const existingColor = jars?.some((jar) => {
       return jar.color === color;
     });
-
-    if (existingColor) {
+    // temporary solution. add an event handler when editing
+    if (existingColor && !isEditMode) {
       setErrorText('Цей колір вже зайнятий 😔 Спробуй обрати інший 😉');
       return;
     }
 
+    // continue here
     const createJarPayload: CreateJarPayload = {
       url,
       ownerName: owner,
       fundraisingCampaignId: Number(fundraisingId),
       color: color,
+      goal: goal ? Number(goal) : null,
+      otherSourcesAccumulated: Number(otherIncoms),
+      isFinished: isFinished === 'true',
     };
 
-    const response = await postJar(createJarPayload);
-
-    addJar(response);
+    if (isEditMode) {
+      const response = await Promise.resolve({
+        ...jar,
+        ...getPayload(createJarPayload, jar),
+      });
+      replaceJar(response);
+    } else {
+      const response = await postJar(createJarPayload);
+      addJar(response);
+    }
+    console.log(createJarPayload);
     resetForm();
     closeDialog();
   };
 
   return (
     <Dialog
-      title='Давай додамо баночку!'
+      title={
+        isEditMode ? 'Давай відредагуємо баночку!' : 'Давай додамо баночку!'
+      }
       dialogState={dialogState}
       renderButton={() => (
-        <Button className={classNames(styles['add-jar'])} onClick={openDialog}>
-          ➕ Додати банку
+        <Button
+          className={`
+        ${isEditMode ? styles['edit-jar'] : styles['add-jar']}
```

Comment:
Not sure why you need this one. Difference is in `:hover` pseudoclass? Why not use same stylings for both of them